src/pages/signup-page/Signup.jsx

```diff
@@ -20,70 +20,139 @@ const [state, dispatch] = useReducer(AuthReducer, {
   const { setLogin } = useContext(LoginContext);
   const signupHandler = async (e) => {
     e.preventDefault();
-    if(state.field.name && state.field.email && state.field.password){
-    try {
-      const response = await axios.post(`/api/auth/signup`, state.field);
-      if (response.status === 201) {
-        setLogin(true);
-        setToastState(true);
-        setToastMsg("Signin sucessfully");
-        setTimeout(() => {
-          setToastState(false);
-        }, 1500);
+    if (
+      state.field.email.indexOf("@") >= 0 &&
+      state.field.password.length >= 8 &&
+      state.field.confirmPassword === state.field.password
+    ) {
+      try {
+        const response = await axios.post(`/api/auth/signup`, state.field);
+        if (response.status === 201) {
+          setLogin(true);
+          setToastState(true);
+          setToastMsg("Signin sucessfully");
+          setTimeout(() => {
+            setToastState(false);
+          }, 1500);
+        }
+               
+        localStorage.setItem("token", response.data.encodedToken);
+      } catch (error) {
+        console.log(error);
       }
-      // saving the encodedToken in the localStorage
-      localStorage.setItem("token", response.data.encodedToken);
-    } catch (error) {
-      console.log(error);
     }
-  }
+
+    if (state.field.email.indexOf("@") === -1) {
+      dispatch({ type: "EMAIL_ERR" });
+    } else {
+      dispatch({ type: "EMAIL_ERR" });
+    }
+
+    if (state.field.password.length < 8) {
+      dispatch({ type: "PASSWORD_ERR" });
```

Comment:
Both if & else condition are dispatching same action. Why is that so?

Reply:
so basically it show err, here i m using toggle, so if user correctly input, the same condition turns false and err will be removed